sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/StorageSharedKeyCredential.java

```diff
@@ -134,6 +153,13 @@ private String buildStringToSign(URL requestURL, String httpMethod, Map<String,
             getStandardHeaderValue(headers, "Range"),
             getAdditionalXmsHeaders(headers),
             getCanonicalizedResource(requestURL));
+
+        if (logStringToSign) {
+            StorageImplUtils.logStringToSign(logger, stringToSign,
+                new Context(Constants.STORAGE_LOG_STRING_TO_SIGN, true));
```

Comment:
Let's make this a constant so we don't need to instantiate a new instance every time this code path is executed